error/s2n_errno.h

```diff
@@ -195,12 +203,12 @@ typedef enum {
     S2N_ERR_INVALID_DYNAMIC_THRESHOLD,
     S2N_ERR_INVALID_ARGUMENT,
     S2N_ERR_NOT_IN_UNIT_TEST,
-    S2N_ERR_BAD_KEY_SHARE,
```

Comment:
Are we intentionally change the error category?

Reply:
Yeah, there are some errors which have incorrect type. I fixed this one, but there are more. Opened #1277 to fix this in future.